src/app/tabs/newAdmin/addresses/operations.js

```diff
@@ -103,6 +122,57 @@ const setMultiChainAddress = (domain, chainId, address, isNew) => async (dispatc
   );
 };
 
+/**
+ * Sets values in the Definitive Resolver.
+ * @param {string} domain the domain to be set
+ * @param {uint8} chainId chain Id for the coin
+ * @param {address} address address to be set
+ * @param {bool} isNew is this a new address?
+ */
+const setDefinitiveAddress = (domain, chainId, address, isNew) => async (dispatch) => {
+  const chainName = getChainNameById(chainId);
+  const chainIndex = getIndexById(chainId);
+  const accounts = await window.ethereum.enable();
+  const currentAddress = accounts[0];
+
+  dispatch(requestSetChainAddress(chainName));
+
+  // encode value if it is not empty:
+  let encodeValue = address;
+  if (encodeValue !== EMPTY_ADDRESS) {
+    try {
+      encodeValue = formatsByCoinType[chainIndex].decoder(address);
+    } catch (error) {
+      return dispatch(errorSetChainAddress(chainName, ADDRESS_ENCODING_ERROR, address));
+    }
+  }
+
+  return definitiveResolver.methods.setAddr(namehash(domain), chainId, encodeValue)
+    .send({ from: currentAddress }, (error, result) => {
+      dispatch(waitingSetChainAddress(chainName));
+      if (error) {
+        return dispatch(errorSetChainAddress(chainName, error.message));
+      }
+
+      const transactionConfirmed = () => () => {
```

Comment:
```suggestion
      const transactionConfirmed = () => {
```

Reply:
Sadly this is needed but is high on my list of things to refactor from Retrium.

When I originally wrote the `transsactionListener` class I modeled it after the notifications bar class. So, in the class, I used `dispatch(callback)` which was incorrect for this class:

https://github.com/rnsdomains/rns-manager-react/blob/develop/src/app/helpers/transactionListener.js#L9

I should have just done `callback()` and then there would be no need for the extra `() => () =>` because `dispatch` isn't being sent. It doesn't need to be sent because it already exists in the operations file.